File: Candy/candy.dart
class Solution {
  int candy(List<int> ratings) {
    final List<int> candies = List.filled(ratings.length, 1, growable: false);

    for (var i = 1; i < ratings.length; i++) {
      if (ratings[i - 1] < ratings[i]) {
        candies[i] = candies[i - 1] + 1;
      }
    }

    for (var i = ratings.length - 2; i >= 0; i--) {
      if (ratings[i + 1] < ratings[i] && candies[i + 1] >= candies[i]) {
        candies[i] = candies[i + 1] + 1;
      }
    }

    return candies.reduce((value, element) => value + element);
  }
}

File: Candy/candy.js
/**
 * @param {number[]} ratings - The rate arrays of child
 * @return {number}
 */
var candy = function (ratings) {
	let candies = new Array(ratings.length).fill(1);

	for (let i = 1; i < ratings.length; i++) {
		if (ratings[i - 1] < ratings[i]) {
			candies[i] = candies[i - 1] + 1;
		}
	}

	for (let i = ratings.length - 2; i >= 0; i--) {
		if (ratings[i + 1] < ratings[i] && candies[i + 1] >= candies[i]) {
			candies[i] = candies[i + 1] + 1;
		}
	}

	return candies.reduce((acc, cur) => acc + cur, 0);
};
